src/Analyzers/Core/Analyzers/RemoveUnusedMembers/AbstractRemoveUnusedMembersDiagnosticAnalyzer.cs

```diff
@@ -486,7 +510,10 @@ private void OnSymbolEnd(SymbolAnalysisContext symbolEndContext, bool hasUnsuppo
                             break;
 
                         case IPropertySymbol property:
-                            if (property.GetMethod != null && property.SetMethod != null)
+                            // We change the message only if both 'get' and 'set' accessors are present and
+                            // there are no shadow 'get' accessor usages. Otherwise the message will be confusing
+                            if (property.GetMethod != null && property.SetMethod != null &&
+                                !_propertiesWithShadowGetAccessorUsages.Contains(property))
```

Comment:
I would make `_propertiesWithShadowGetAccessorUsages` a `HashSet` instead of a `List` because the usage is to check whether it contains a value, but that's mostly just for clarity, because the number of properties will never likely be so high that it would become an issue.